scripts/pylint_extensions.py

```diff
@@ -1383,6 +1384,60 @@ def process_module(self, node):
                 is_docstring = False
 
 
+class SpaceBelowFileOverviewChecker(checkers.BaseChecker):
+    """Checks if there is an empty line between file overview and imports."""
+
+    __implements__ = interfaces.IRawChecker
+    name = 'space_between_imports_and_file-overview'
+    priority = -1
+    msgs = {
+        'C0024': (
+            'No empty line used between file overview and import.',
+            'no-empty-line-provided-below-fileoverview',
+            'please provide a line between file overview and imports.'
+        ),
+        'C0025': (
+            'Single empty line should be provided above imports.',
+            'only-a-single-empty-line-should-be-provided',
+            'please provide a single new line above imports.'
+        )
+    }
+
+    def process_module(self, node):
+        """Process a module to ensure that there is an empty line between
+        file overview and imports.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node to access module content.
+        """
+
+        multi_line_indicator = b'"""'
+        file_content = read_from_node(node)
+        file_length = len(file_content)
+        triple_quote_counter = 0
+        for line_num in python_utils.RANGE(file_length):
+            line = file_content[line_num].strip()
+            # Single line comment, ignore it.
+            if line.startswith(b'#'):
+                continue
+            triple_quote_counter += line.count(multi_line_indicator)
+            previous_line = file_content[line_num - 1].strip()
+            if (line.startswith(b'import') or line.startswith(b'from')):
```

Comment:
The logic here seems wrong. Once you've encountered the fileoverview line closing quote and are on the next line, just check whether that next line is empty. The check should be independent of the contents of that line.

You can also check for the longest consecutive run of empty or whitespace-only lines starting at that point and verify that the count of lines in that run is equal to 1.

Hopefully this helps to simplify the logic.

Also, make sure to consider the case in which the file doesn't have a fileoverview comment, and the first triple-quote comment encountered is in a class/method docstring. We should detect that the fileoverview is missing. I'm not sure if there's an existing pylint check in place for this -- if so, you can just mention that here so that it's clear what assumptions you are working with.

Reply:
> We should detect that the fileoverview is missing

@seanlip, there is an inbuilt checker by Pylint for missing file overview. But if I add a test case having no fileoverview it can't detect it.
So should we still add a test case for testing this?

Reply:
First, see if you can find out why it doesn't detect it (in general, this is always a good practice -- figure out what's wrong before you start modifying things). Based on the resulting analysis, we can figure out what to do next.

Reply:
> if so, you can just mention that here

Sorry I didn't completely understand what you meant here

Reply:
I meant mentioning it in a comment.

Reply:
> see if you can find out why it doesn't detect it

I guess in a test case only the linter being tested is activated .
For example in my case only `SpaceBelowFileOverviewChecker` is activated
`     self.checker_test_object.CHECKER_CLASS = (
            pylint_extensions.SpaceBelowFileOverviewChecker)
`
Pls correct me If I'm missing something

Reply:
@seanlip, So should we still add the test case having no file overview?

Reply:
> I guess in a test case only the linter being tested is activated .

What checks have you done to verify this statement? Please describe how you came to that conclusion (it needs to be more than a guess, it should be supported by evidence).

Reply:
@seanlip, I have verified the above by changing the `self.checker_test_object.CHECKER_CLASS` at the beginning of the testing class.
Example: in the case when I register `SpaceBelowFileOverviewChecker` as `CHECKER_CLASS` and give excessive lines between  function definitions(which is detected by `ExcessiveEmptyLinesChecker` class) , the excess lines are not detected.
But when I register `ExcessiveEmptyLinesChecker` as the `CHECKER_CLASS`, It detects the excess lines.

Reply:
That's good. (Also, in the future, please include the analysis as a matter of course when stating any conclusions.)

There is only one check left to do: can you verify that pylint itself catches missing fileoverviews when the full linting process is done? If so then you can go ahead with what I said above: leave a comment explaining the assumption you're making about the existence of such a fileoverview (or, to be more rigorous, you can check that the corresponding rule isn't excluded in .pylintrc), and you don't need to add a test here with missing fileoverview since the assumption underlying everything here is that the fileoverview exists.

Reply:
@seanlip, yes I verified that Pylint does throw a message if there is no file overview.
Although the modified code works as expected even if there is no file-overview.

Reply:
Hm ... when you say "the modified code works as expected even if there is no file-overview", what do you mean exactly? I.e. what is the "expected behaviour" if there's no file overview?

(For me this case is "undefined" since we are assuming, by virtue of the other pylint check, that a fileoverview will always be present.)

Reply:
I added an extra check for not mistaking the triple quote in Args of method/class as file overview.(As this happens only when there is no file overview I'll remove this check)

Reply:
Sorry for the confusion